### PR TITLE
bluetooth: hci: enable `BT_SPI` from DT

### PR DESCRIPTION
**Pull request by JordanYates**
Add dedicated Kconfig symbols for the two SPI backend variants. This
allows enabling the symbols by default from devicetree, and cleans up
the cmake logic.

The `HCI_SPI` symbol is now automatically selected from devicetree,
removing the need for board specific defaults.

### CODE REVIEW

**Comment by jhedberg** [comment]
Not sure if this is quite correct, due to this hackery in `CMakeLists.txt` (which I'd gladly get rid of, but it may require some refactoring somewhere):
https://github.com/zephyrproject-rtos/zephyr/blob/1aaea30adb62df1aff2ccb167966db443667d5f6/drivers/bluetooth/hci/CMakeLists.txt#L20-L26

**Comment by JordanYates** [comment]
> Not sure if this is quite correct, due to this hackery in `CMakeLists.txt` (which I'd gladly get rid of, but it may require some refactoring somewhere):
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/1aaea30adb62df1aff2ccb167966db443667d5f6/drivers/bluetooth/hci/CMakeLists.txt#L20-L26

Yea noticed this after pushing, working on it now

**Review by jhedberg — DISMISSED** [review]

File: drivers/bluetooth/hci/Kconfig [hunk 1]
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
...
 	  additional platform specific knowledge may need to be added as
 	  devices are.
 
+config BT_SPI_INIT_PRIORITY
+	int "BT SPI init priority"
+	depends on BT_SPI
+	default 75
+
+config BT_SPI_ZEPHYR
+	bool "SPI HCI as implemented by Zephyr"
```
Comment (HoZHel) [inline drivers/bluetooth/hci/Kconfig:66]:
```suggestion
	bool "HCI over SPI as implemented by Zephyr"
```

File: drivers/bluetooth/hci/Kconfig [hunk 2]
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
...
+	default y
+	depends on DT_HAS_ZEPHYR_BT_HCI_SPI_ENABLED
+	select BT_SPI
+	help
+	  Bluetooth HCI driver for communication with Bluetooth controllers
```
Comment (HoZHel) [inline drivers/bluetooth/hci/Kconfig:71]:
```suggestion
	  Bluetooth HCI over SPI driver for communication with Bluetooth controllers
```

File: drivers/bluetooth/hci/Kconfig [hunk 3]
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
...
+	bool "SPI HCI as implemented by ST BlueNRG"
+	default y
+	depends on DT_HAS_ST_HCI_SPI_V1_ENABLED || DT_HAS_ST_HCI_SPI_V2_ENABLED
+	select BT_SPI
+	help
+	  Bluetooth HCI driver for communication with the BlueNRG-1 and BlueNRG-2
+	  devices from ST Microelectronics.
+
+config BT_BLUENRG_ACI
+	bool "ACI message with BlueNRG-based devices"
```
Comment (HoZHel) [inline drivers/bluetooth/hci/Kconfig:84]:
```suggestion
	bool "ACI configuration message with BlueNRG-based devices"
```

File: drivers/bluetooth/hci/Kconfig [hunk 4]
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
...
+	  Bluetooth HCI driver for communication with Bluetooth controllers
+	  running Zephyr.
+
+config BT_SPI_BLUENRG
+	bool "SPI HCI as implemented by ST BlueNRG"
+	default y
+	depends on DT_HAS_ST_HCI_SPI_V1_ENABLED || DT_HAS_ST_HCI_SPI_V2_ENABLED
+	select BT_SPI
+	help
+	  Bluetooth HCI driver for communication with the BlueNRG-1 and BlueNRG-2
```
Comment (HoZHel) [inline drivers/bluetooth/hci/Kconfig:80]:
```suggestion
	  Bluetooth HCI over SPI driver for communication with STMicroelectronics radio devices.
```

File: drivers/bluetooth/hci/Kconfig [hunk 5]
```diff
@@ -48,15 +48,46 @@ config BT_HCI_IPC
...
+	  devices from ST Microelectronics.
+
+config BT_BLUENRG_ACI
+	bool "ACI message with BlueNRG-based devices"
+	depends on BT_SPI_BLUENRG
+	select BT_HCI_SET_PUBLIC_ADDR
+	help
+	  Enable support for devices compatible with the BlueNRG Bluetooth
+	  Stack. Current driver supports: ST BLUENRG-MS.
```
Comment (HoZHel) [inline drivers/bluetooth/hci/Kconfig:89]:
```suggestion
	  Enable vendor specific commands to switch radio to controller-only mode.
```

**Review by HoZHel — CHANGES_REQUESTED** [review]

**Comment by JordanYates** [comment]
@HoZHel the changes in this PR are not modifying the form of any of the Kconfig text, just rearranging the symbols (And duplicating in the case of `SPI_HCI_ZEPHYR`).
I think substantive changes should be their own PR, so that all the symbols can be addressed at once/wording changes discussed independently.

**Comment by HoZHel** [comment]
> @HoZHel the changes in this PR are not modifying the form of any of the Kconfig text, just rearranging the symbols (And duplicating in the case of `SPI_HCI_ZEPHYR`). I think substantive changes should be their own PR, so that all the symbols can be addressed at once/wording changes discussed independently.

Well, `config BT_SPI_BLUENRG` is not an rearrangement and it has a wrong definition.

**Comment by JordanYates** [comment]
Removed prompts from the new symbols as done in #77418

**Review by jhedberg — DISMISSED** [review]

**Comment by msmttchr** [comment]
@JordanYates, recently PR #71352 added a new HCI board which possibly needs update before merging this PR.

**Comment by JordanYates** [comment]
> @JordanYates, recently PR #71352 added a new HCI board which possibly needs update before merging this PR.

Done

**Review by jhedberg — DISMISSED** [review]

**Comment by msmttchr** [comment]
> > @HoZHel the changes in this PR are not modifying the form of any of the Kconfig text, just rearranging the symbols (And duplicating in the case of `SPI_HCI_ZEPHYR`). I think substantive changes should be their own PR, so that all the symbols can be addressed at once/wording changes discussed independently.
> 
> Well, `config BT_SPI_BLUENRG` is not an rearrangement and it has a wrong definition.

I think this comment needs also to be addressed before merging this PR, since the information provided can be misleading.

**Comment by msmttchr** [comment]
> The `BT_SPI_BLUENRG` symbol was updated mostly in line with the request
Ah, sorry I missed that.

**Review by HoZHel — DISMISSED** [review]

**Comment by JordanYates** [comment]
Rebasing on main in the hope the unrelated CI failures are resolved

**Comment by JordanYates** [comment]
Rebase to resolve release notes conflict

**Review by jhedberg — APPROVED** [review]

**Review by hermabe — APPROVED** [review]

**Comment by jhedberg** [comment]
@erwango any idea about the CI failures here? They don't seem to be related to this PR, rather something's off with mbedtls and wifi on ST platforms. Is that a know issue?

**Comment by erwango** [comment]
> @erwango any idea about the CI failures here? They don't seem to be related to this PR, rather something's off with mbedtls and wifi on ST platforms. Is that a known issue?

Also failing with `nrf9160dk/nrf9160/ns`, so I guess it's rather an issue with the test itself.

**Comment by jhedberg** [comment]
> Also failing with `nrf9160dk/nrf9160/ns`, so I guess it's rather an issue with the test itself.

@erwango where do you see that? I only see `stm32l562e_dk/stm32l562xx/ns` failing in the CI logs.

**Comment by erwango** [comment]
> > Also failing with `nrf9160dk/nrf9160/ns`, so I guess it's rather an issue with the test itself.
> 
> @erwango where do you see that? I only see `stm32l562e_dk/stm32l562xx/ns` failing in the CI logs.

Reproduced locally. But I don't know why only `stm32l562e_dk/stm32l562xx/ns` is tested.

**Comment by jhedberg** [comment]
The CI failure should get fixed by #80381

**Pull request merged**